### ISSUE TITLE
lsd.cpp:90:25: error: ‘uintptr_t’ is not a member of ‘boost’   , m_cookie((random() ^ boost::uintptr_t(this)) & 0x7fffffff)

### ISSUE THREAD

**Issue by NetSharkDon** [arvidn/libtorrent#1516]
HI 
i build libtorrent on centos 
with 
_Package boost-1.53.0-26.el7.x86_64 already installed and latest version
Package boost-devel-1.53.0-26.el7.x86_64 already installed and latest version_
but ERROR
 ‘uintptr_t’ is not a member of ‘boost’
boost 1.53
Someone Can Help Me  ?Thanks

**Comment by Falcosc** [comment]
You need alot more boost libs. Please try to find the lib where the missing member is included or install all boost libs like I did.

**Comment by NetSharkDon** [comment]
i have 
_yum install boost*_  but it doesnt work
how did you _install all boost libs like I did_
thanks

**Comment by NetSharkDon** [comment]
lsd.cpp: In constructor ‘libtorrent::lsd::lsd(libtorrent::io_service&, const peer_callback_t&)’:
lsd.cpp:90:25: error: ‘uintptr_t’ is not a member of ‘boost’
  , m_cookie((random() ^ boost::uintptr_t(this)) & 0x7fffffff)
                         ^
lsd.cpp:90:25: note: suggested alternative:
In file included from /usr/lib/gcc/x86_64-redhat-linux/4.8.5/include/stdint.h:9:0,
                 from /usr/include/inttypes.h:27,
                 from ../include/libtorrent/config.hpp:75,
                 from ../include/libtorrent/socket.hpp:36,
                 from ../include/libtorrent/lsd.hpp:36,
                 from lsd.cpp:33:
/usr/include/stdint.h:122:27: note:   ‘uintptr_t’
 typedef unsigned long int uintptr_t;
                           ^
make[1]: *** [lsd.lo] Error 1
make[1]: Leaving directory `/root/libtorrent-rasterbar-1.1.1/src'
make: *** [all-recursive] Error 1

**Comment by Falcosc** [comment]
and your gcc is to old, you need at least 5.1
Please upgrade your dist to get the new gcc, maybe you get the missing boost libs after the upgrade as well.

**Comment by arvidn** [comment]
libtorrent 1.1.x does not require C++11, and since this is referring to boost, I imagine this is not building master. If it's a matter of a missing header file, I would expect there to be another error before this one, indicating the boost header can't be found.
@NetSharkDon which version of libtorrent are you building?
is this part of a package?
perhaps this ``lsd.cpp`` is missing the include ``<boost/cstdint.hpp>``

**Comment by NetSharkDon** [comment]
@arvidn 
Version libtorrent 1.1.1
there are not errors before this one,

**Comment by NetSharkDon** [comment]
Version  libtorrent 1.2.0.0
  _CXX      web_connection_base.lo
In file included from ../include/libtorrent/settings_pack.hpp:36:0,
                 from ../include/libtorrent/aux_/session_settings.hpp:37,
                 from ../include/libtorrent/peer_connection.hpp:55,
                 from ../include/libtorrent/web_connection_base.hpp:42,
                 from web_connection_base.cpp:38:
../include/libtorrent/entry.hpp:300:3: error: 'aligned_union' in namespace 'std' does not name a type
   std::aligned_union<1
   ^
../include/libtorrent/entry.hpp: In constructor 'libtorrent::entry::entry(U)':
../include/libtorrent/entry.hpp:150:9: error: 'data' was not declared in this scope
    new(&data) string_type(std::move(v));
         ^
../include/libtorrent/entry.hpp: In member function 'libtorrent::entry& libtorrent::entry::operator=(U)':
../include/libtorrent/entry.hpp:191:9: error: 'data' was not declared in this scope
    new(&data) string_type(std::move(v));
         ^
make[1]: *** [web_connection_base.lo] Error 1
make[1]: Leaving directory `/root/libtorrent-master/src'_

**Comment by NetSharkDon** [comment]
i use BOOST 1.53

**Comment by Falcosc** [comment]
aligned_union is a new c++11 feature which need at least gcc 5.1

**Comment by arvidn** [comment]
and I suppose boost::uintptr_t wasn't added until boost 1.54 then either. If you want to hack it to get the build going, you could probably try replacing ``boost::uintptr_t`` with just ``uintptr_t`` or ``std::uintptr_t``.

However, with such old setup, you may run into other issues too

**Comment by Sahl-A** [comment]
The mentioned Issue by @NetSharkDon still persists. 
- I use CentOs 7 GNOME Version 3.14.2
- I downloaded the source Tarball "libtorrent-rasterbar-1.1.3.tar.gz" from https://github.com/arvidn/libtorrent/releases
- I use boost-1.53.0-26.el7.x86_64 
- I have all the required tools; "qt-devel" "boost-devel" "openssl-devel" "qt5-qtbase-devel" "qt5-linguist" "gcc" "gcc-c++" "kernel-devel"
- ./configure runs smoothly without any errors. 
- $ make provides the following error:

lsd.cpp: In constructor ‘libtorrent::lsd::lsd(libtorrent::io_service&, const peer_callback_t&)’:
lsd.cpp:90:25: error: ‘uintptr_t’ is not a member of ‘boost’

For the full error, please check the attached screenshot below:

![screenshot](https://cloud.githubusercontent.com/assets/27885879/25304240/e0d07544-276b-11e7-928f-8251b0345e7a.jpg)
#
@arvidn I would appreciate your help

**Comment by arvidn** [comment]
``boost::uintptr_t`` should be defined in ``<boost/cstdint.hpp>``, you may want to double check that file is included. You may also want to take a look at your copy of that file to see if there's some compile time switch that disables it.

``std::uintptr_t`` was introduced in c++11, so if your compiler supports that, you could try replacing ``boost`` with ``std`` for this type.

**Comment by arvidn** [comment]
you could most likely also use ``long long`` instead

**Comment by f100024** [comment]
@Sahl-A could you provide your gcc version?

**Comment by Sahl-A** [comment]
@f100024 the yum output>>>> Package gcc-4.8.5-11.el7.x86_64 already installed and latest version
Nothing to do

@arvidn after I double checked, **_boost::uintptr_t_** is not defined in `<boost/cstdint.hpp>` so, this is no time switch to check. 

How would replace boost with std. How also would I use long long.

Thank you both for your advise

**Comment by arvidn** [comment]
actually, there are other places in libtorrent that use ``uintptr_t``, just not from the boost namespace. This relies on (non-standard) extensions, which presumably more relevant compilers implement (from C99 I believe).

Please try to just remove the ``boost::`` prefix before ``uintptr_t``.

If this works for you, I will apply this patch for the 1.1.4 release.

**Comment by arvidn** [comment]
I made this change in RC_1_1

**Comment by f100024** [comment]
as solution try use newer version of gcc:
install `yum install devtoolset-6` and enable it  `scl enable devtoolset-6 bash`

**Comment by Sahl-A** [comment]
@f100024 Thank you for your help but it did not work.

@arvidn sorry for my late response. I took some time for me to understand what RC_1_1 and what is branches and what is git. I tried to replace all the files from RC_1_1 to the master branch. However, this didn't work as I faced an error after running ./configure. Then I replaced only the modified files from RC_1_1 branch to the master branch and everything went smoothly without any errors. 

Thank you for your effort. Finally, I was able to compile this without any errors.
